packages/gestalt/src/InfoButton.js

```diff
@@ -0,0 +1,147 @@
+  /**
+   * Callback fired when Link is clicked (pressed and released) with a mouse or keyboard. See [OnLinkNavigationProvider](https://gestalt.pinterest.systems/web/utilities/onlinknavigationprovider) to learn more about link navigation.
+   */
+  onClickLink?: ({|
+    event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+    dangerouslyDisableOnNavigation: () => void,
+  |}) => void,
+  /**
+   * An object representing the zIndex value of the Dropdown menu. Learn more about [zIndex classes](https://gestalt.pinterest.systems/web/zindex_classes)
+   */
+  zIndex?: Indexable,
+|};
+
+/**
+ * [InfoButton](https://gestalt.pinterest.systems/web/infobutton) is a turnkey solution to provide help/guidance for an element on the screen.
+ */
+function InfoButton({
+  accessibilityLabel,
+  text,
+  accessibilityLinkLabel,
+  linkHref,
+  linkText,
+  idealDirection,
+  onClickLink,
+  zIndex,
+}: Props): Node {
+  const ref = useRef<?HTMLElement>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const { accessibilityDefaultTooltipMessage, accessibilityDefaultLinkLabel, accessibilityIcon } =
+    useDefaultLabelContext('InfoButton');
+
+  const toggleView = () => {
+    setIsOpen(!isOpen);
+    if (isOpen) {
+      ref.current?.focus();
+    }
+  };
+
+  return (
+    <Fragment>
+      <TapArea
+        accessibilityExpanded={isOpen}
+        accessibilityControls="info-dialog"
+        accessibilityLabel={accessibilityDefaultTooltipMessage}
+        fullWidth={false}
+        onTap={toggleView}
+        ref={ref}
+        role="button"
+        rounding="circle"
+      >
+        <Tooltip text={accessibilityDefaultTooltipMessage}>
+          <div className={styles.infoButton}>
+            <Icon
+              accessibilityLabel={accessibilityIcon}
+              color="inverse"
+              icon="question-mark"
+              size={8}
+            />
+          </div>
+        </Tooltip>
+      </TapArea>
+      {isOpen && (
+        <Layer zIndex={zIndex}>
+          <Popover
+            id="info-dialog"
+            accessibilityLabel={accessibilityLabel ?? accessibilityDefaultLinkLabel}
+            anchor={ref.current}
+            onDismiss={toggleView}
+            idealDirection={idealDirection ?? 'down'}
+            positionRelativeToAnchor={false}
+          >
+            <TrapFocusBehavior>
+              <Box padding={4} rounding={4} minWidth={230} tabIndex={0}>
+                <Text align="center">{text}</Text>
+                {linkHref && (
+                  <Box
+                    padding={3}
+                    with="100%"
+                    display="flex"
+                    alignItems="center"
+                    justifyContent="center"
+                    marginTop={3}
+                  >
+                    <Text weight="bold" size="300">
+                      <Link
+                        accessibilityLabel={accessibilityLinkLabel ?? accessibilityDefaultLinkLabel}
```

Comment:
`accessibilityLabel` is optional on Link, and we're making `link.text` required, so we don't need a default label.